OpenDreamRuntime/Objects/MetaObjects/DreamMetaObjectIcon.cs

```diff
@@ -0,0 +1,126 @@
+using System.Collections.Generic;
+using System.IO;
+using OpenDreamRuntime.Procs;
+using OpenDreamRuntime.Resources;
+using OpenDreamShared.Dream;
+using OpenDreamShared.Resources;
+
+namespace OpenDreamRuntime.Objects.MetaObjects {
+    sealed class DreamMetaObjectIcon : DreamMetaObjectDatum
+    {
+        [Dependency] private readonly DreamResourceManager _rscMan = default;
+        public struct DreamIcon {
+            public DreamResource Resource;
+            public string? State;
+            public AtomDirection Direction;
+            public int Frame;
+            public byte? Moving;
+            public DMIParser.ParsedDMIDescription Description;
+
+            public DreamIcon(DreamResource rsc, DreamValue? state, DreamValue? dir, DreamValue? frame, DreamValue? moving)
+            {
+                Resource = rsc;
+                Description = DMIParser.ParseDMI(new MemoryStream(rsc.ResourceData));
+
+                // TODO confirm BYOND behavior of invalid types
```

Comment:
For the 'no name' empty icon state, it's an empty string. Default frames and dirs is 1. Moving default is null.
```
# BEGIN DMI
version = 4.0
	width = 32
	height = 192
state = "heavenlight"
	dirs = 1
	frames = 1
state = ""
	dirs = 1
	frames = 1
# END DMI
```
for 
![image](https://user-images.githubusercontent.com/4741640/160770287-be0577ec-701c-45b3-a152-275c7a5b1b78.png)

Interesting tidbit: All trailing and preceding spaces in an icon state are invalid, and trimmed in DM.